client/firecracker_client_factory.go

```diff
@@ -0,0 +1,43 @@
+package client
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/sirupsen/logrus"
+
+	httptransport "github.com/go-openapi/runtime/client"
+)
+
+// NewHttpClientForUNIXSocket creates a new firecracker HTTP client configured at the specified a Unix socketPath.
+func NewHttpClientForUNIXSocket(socketPath string, logger *logrus.Entry, debug bool) *Firecracker {
```

Comment:
My take on that was, if anyone will care about a "UnixSocketTransport" or will they care more about a ready to use client. The difference between the two being formats can be passed explicitly with the suggestion above and a consumer can use sets in the unixTransport (which we already expose arguments).  
   
Another reason was, that it exposes clients to OpenAPIs: `runtime.Client` which didn't seem very nice if we go by the idea that we should keep the surface simpler and easy to use.

Reply:
> Not sure I follow since this returns a http.Transport and creates the same client to begin with

The return type is: [*Runtime](https://github.com/go-openapi/runtime/blob/master/client/runtime.go#L200)  

The New accepts the interface: [runtime.ClientTransport](https://github.com/go-openapi/runtime/blob/master/client_operation.go#L38)

> It'll simplify the function and prevent the function from doing too much.

I agree with the idea of not having the client do too much, in this case the only extra thing `NewHttpClientForUNIXSocket` does over the proposed `NewUnixSocketTransport` is: `return client.New(unixTransport, strfmt.NewFormats())`  

I can update the PR based on whatever makes more sense to expose as a function, one which returns a UnixTransport or a UnixClient or both

Reply:
Yea, I prefer having this be just `NewUnixSocketTransport`. Again, thank you for taking the time to do this and whenever you get the chance, please update.

Reply:
Sure! Updated